internal/local/structure_v2.go

```diff
@@ -601,19 +607,35 @@ func writeDataSources(rootDir string, dataSources []DataSourceStructure) error {
 			); err != nil {
 				return err
 			}
-			if err := WriteFile(
-				filepath.Join(ruleDir, FileSchema.String()),
-				0666,
-				bytes.NewReader(dataSchema),
-			); err != nil {
-				return err
+
+			schema := rule[NameSchema]
+			if schema != nil {
+				dataSchema, err := MarshalJSON(schema)
```

Comment:
[meganit/naming] could in theory tighten the scope of this named `var` to just `data` now that we're in this `okSchema` closure